src/java/main/org/apache/zookeeper/server/ReferenceCountedACLCache.java

```diff
@@ -109,16 +109,18 @@ public synchronized void deserialize(InputArchive ia) throws IOException {
             }
             List<ACL> aclList = new ArrayList<ACL>();
             Index j = ia.startVector("acls");
-            while (!j.done()) {
-                ACL acl = new ACL();
-                acl.deserialize(ia, "acl");
-                aclList.add(acl);
-                j.incr();
+            if (j != null) {
```

Comment:
- can we have a more elegant way to process this NPE ?
- BTW. incorrect commit message.   `git commit --amend -m "your new message"` to modify it

Reply:
just as @anmolnar  review, we should throw exception if j == null, so add code just like 

if (j == null){
    throw Exception(error_message);
}
@maoling can this change be elegant?